### PR TITLE
Adding runtime property for redeployment and K8

### PR DESCRIPTION
**Pull request by lmanohara**

### CODE REVIEW

File: modules/jaggeryapps/appmgt/src/modules/application/application.jag [hunk 1]
```diff
@@ -359,17 +364,8 @@ function getContainers(tenantDockerRegistryUrl, applicationName, appServiceName,
     container1.setContainerName(applicationName);
```
Comment (punnadi) [inline modules/jaggeryapps/appmgt/src/modules/application/application.jag]: Rename 'container1' to a meaningful value

File: modules/jaggeryapps/appmgt/src/modules/application/application.jag [hunk 2]
```diff
@@ -359,17 +364,8 @@ function getContainers(tenantDockerRegistryUrl, applicationName, appServiceName,
...
     container1.setBaseImageName(appBaseImageName);
     container1.setBaseImageVersion(applicationRevision);
-    //setting runtime properties to the container
-    if(runtimeProperties){
-        var envs1 = new java.util.HashMap();
-        var runtimePropertyIterator = runtimeProperties.iterator();
-        while(runtimePropertyIterator.hasNext()){
-            var runtimeProperty = runtimePropertyIterator.next();
-            envs1.put(runtimeProperty.getPropertyName(), runtimeProperty.getPropertyValue());
```
Comment (punnadi) [inline modules/jaggeryapps/appmgt/src/modules/application/application.jag]: Rename 'envs1'

File: modules/jaggeryapps/appmgt/src/modules/application/application.jag [hunk 3]
```diff
@@ -359,17 +364,8 @@ function getContainers(tenantDockerRegistryUrl, applicationName, appServiceName,
...
-        }
-        envs1.put("ORG","WSO2");
-        container1.setEnvVariables(envs1);
-    }
+    var envs = getRuntimeProperties(runtimeProperties);
+    container1.setEnvVariables(envs);
     var serviceProxyList1 = new java.util.ArrayList();
     var serviceProxy1 = new Packages.org.wso2.appcloud.provisioning.runtime.beans.ServiceProxy();
```
Comment (punnadi) [inline modules/jaggeryapps/appmgt/src/modules/application/application.jag]: Rename 'serviceProxyList1'

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java [hunk 4]
```diff
@@ -1211,6 +1211,8 @@ public void addContainer(Container container, int deploymentId) throws AppCloudE
                 container.setImageName(rs.getString("image_name"));
```
Comment (punnadi) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java]: Add to SQL Constants

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java [hunk 5]
```diff
@@ -1151,7 +1151,7 @@ public Deployment getDeployment(int applicationId) throws AppCloudException{
             if(rs.next()) {
                deployment.setDeploymentName(rs.getString("deployment_name"));
                deployment.setReplicas(rs.getInt("replicas"));
-               deployment.setContainers(getContainers(rs.getInt("id")));
+               deployment.setContainers(getContainers(rs.getInt("id"), applicationId));
```
Comment (danushkaf) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java]: Add Constant for "id" explaining what it is. Do we need to add constants to others as well?

File: modules/jaggeryapps/appmgt/src/modules/application/application.jag [hunk 6]
```diff
@@ -359,17 +364,8 @@ function getContainers(tenantDockerRegistryUrl, applicationName, appServiceName,
...
-        var runtimePropertyIterator = runtimeProperties.iterator();
-        while(runtimePropertyIterator.hasNext()){
-            var runtimeProperty = runtimePropertyIterator.next();
-            envs1.put(runtimeProperty.getPropertyName(), runtimeProperty.getPropertyValue());
-        }
-        envs1.put("ORG","WSO2");
-        container1.setEnvVariables(envs1);
-    }
+    var envs = getRuntimeProperties(runtimeProperties);
+    container1.setEnvVariables(envs);
```
Comment (danushkaf) [inline modules/jaggeryapps/appmgt/src/modules/application/application.jag]: Variable name should be meaningful. Why container1?

File: modules/jaggeryapps/appmgt/src/modules/application/application.jag (hunk 1 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/modules/application/application.jag]: This is fixed in master
https://github.com/wso2/app-cloud/blob/330eaf84e4e1f8f99bf76a8990e2b428853cf1d9/modules/jaggeryapps/appmgt/src/modules/application/application.jag#L364

File: modules/jaggeryapps/appmgt/src/modules/application/application.jag (hunk 3 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/modules/application/application.jag]: This is fixed in master
https://github.com/wso2/app-cloud/blob/330eaf84e4e1f8f99bf76a8990e2b428853cf1d9/modules/jaggeryapps/appmgt/src/modules/application/application.jag#L369

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java (hunk 4 above)
Comment (lmanohara) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java]: This is fixed in master code

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java (hunk 5 above)
Comment (lmanohara) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java]: This is fixed in master code

File: modules/jaggeryapps/appmgt/src/modules/application/application.jag (hunk 6 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/modules/application/application.jag]: This is fixed in master code

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java [hunk 7]
```diff
@@ -1279,5 +1281,40 @@ public void addServiceProxy(ServiceProxy serviceProxy, int containerId) throws A
...
+     * Getting runtime properties from database
+     *
+     * @param applicationId id of the application
+     * @return list of runtime properties
+     * @throws AppCloudException
+     */
+    public List<RuntimeProperty> getRuntimeProperties(int applicationId) throws AppCloudException {
+        Connection dbConnection = DBUtil.getDBConnection();
+        PreparedStatement preparedStatement = null;
+        List<RuntimeProperty> runtimeProperties = new ArrayList<>();
```
Comment (danushkaf) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java]: Implementation has to be in generic as well. "new ArrayList<RuntimeProperty>();"

Comment (roshancd) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java]: Line 1294 needs to be :
List<RunntimeProperty> runtimeProperties = new ArrayList<RunntimeProperty>();

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java [hunk 8]
```diff
@@ -1279,5 +1281,40 @@ public void addServiceProxy(ServiceProxy serviceProxy, int containerId) throws A
...
+                RuntimeProperty runtimeProperty = new RuntimeProperty();
+                runtimeProperty.setPropertyName(resultSet.getString(SQLQueryConstants.PROPERTY_NAME));
+                runtimeProperty.setPropertyValue(resultSet.getString(SQLQueryConstants.PROPERTY_VALUE));
+                runtimeProperties.add(runtimeProperty);
+            }
+
+            dbConnection.commit();
+        } catch (SQLException e) {
+            String msg = "Error while getting runtime properties";
+            log.error(msg, e);
```
Comment (prabathabey) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java]: Log and throw is generally considered an anti-pattern. Avoid whenever possible.

File: modules/jaggeryapps/appmgt/src/modules/application/application.jag [hunk 9]
```diff
@@ -383,6 +379,19 @@ function getContainers(tenantDockerRegistryUrl, applicationName, appServiceName,
     return containerList;
 };
 
+function getRuntimeProperties(runtimeProperties){
```
Comment (prabathabey) [inline modules/jaggeryapps/appmgt/src/modules/application/application.jag]: Fix formatting issues in this particular line.

File: modules/jaggeryapps/appmgt/src/modules/application/application.jag [hunk 10]
```diff
@@ -383,6 +379,19 @@ function getContainers(tenantDockerRegistryUrl, applicationName, appServiceName,
...
+    //setting runtime properties to the container
+    var envs = new java.util.HashMap();
+    if(runtimeProperties) {
```
Comment (prabathabey) [inline modules/jaggeryapps/appmgt/src/modules/application/application.jag]: Fix formatting issues in this particular line.

File: modules/jaggeryapps/appmgt/src/modules/application/application.jag (hunk 9 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/modules/application/application.jag]: Yes this has a formatting issues, but this has been refactored in the latest code.

File: modules/jaggeryapps/appmgt/src/modules/application/application.jag (hunk 10 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/modules/application/application.jag]: Yes this has a formatting issues, but this has been refactored in the latest code.

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java (hunk 8 above)
Comment (lmanohara) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/dao/ApplicationDAO.java]: This method has been removed from the latest code. But i created a JIRA to fix this issues in allover the places in code. https://wso2.org/jira/browse/APPCLOUD-72

**Pull request merged**